contents/docs/integrate/ingest-historic-data.mdx

```diff
@@ -0,0 +1,143 @@
+
+- The volume of data
+- API rate limits of both the data source and PostHog
+- Ensure that only the data required for events is exported
+- Handle error scenarios allowing the process to resume form the last successful point
+
+With the above factors in mind it's recommended that you break the process up into steps such as the following:
+
+1. Sequentially export selective data from the data source that represent key events keeping track of where you are in the sequence so that you can restart the process from the last successful point if any problems occur
+2. Store the selected exported data to a new data storage for faster access in future steps
+3. Transform the data to the format you will use with PostHog and again save to a storage mechanism for faster access in later steps.
+
+    The data format should be:
+    ```json
+    {
+        "event": "[event name]",
+        "distinct_id": "[your users' distinct id]",
+        "properties": {
+            "key1": "value1",
+            "key2": "value2"
+        },
+        "timestamp": "[optional timestamp in ISO 8601 format]"
+    }
+    ```
+
+    At this stage it's also important to consider the following:
+    
+    1. Use the same `event` name that you're going to use with your [live data ingestion](/docs/integrate/ingest-live-data) so the historical and live events are seen as the same type within PostHog
+    1. Use the same unique identifier within the `distinct_id` field as you are within your [live data ingestion](/docs/integrate/ingest-live-data) so historical events and live events are associated with the same user
+    1. Convert the old event property names to the new event property names you are using within the events in your [live data ingestion](/docs/integrate/ingest-live-data)
+    1. Ensure that the `timestamp` is a converted version of the original timestamp is ISO 8601 format so that PostHog correctly identifies when the original event occurred
+    1. You may want to set an additional property within `properties` that identifies the original event within the data source
+
+4. Sequentially import the events into PostHog keeping track of the last successfully imported event so that you can restart the process from the last successful point if any problems occur
+
+## Importing events
+
+Once you are ready to import the data into PostHog you can use one of the following:
+
+- a PostHog [server libraries](/docs/integrate/overview#server-libraries)
+- the [PostHog API](/docs/api/overview)
+
+As mentioned above, the data should be in the following format:
+
+```json
+{
+    "event": "[event name]",
+    "distinct_id": "[your users' distinct id]",
+    "properties": {
+        "key1": "value1",
+        "key2": "value2"
+    },
+    "timestamp": "[optional timestamp in ISO 8601 format]"
+}
+```
+
+import NodeCapture from './server/node/snippets/capture.mdx'
+import PythonCapture from './server/python/snippets/capture.mdx'
```

Comment:
Perhaps we can consider in addition writing the whole python file for them (or whatever is easiest/most failsafe to run). E.g. a python script that you can just copy-pasta & provide a JSON file as input with the format above, so we can be closer to 1-click ingestion (listing also the requirements & how to run it which we have in the server library info page, but again the goal of this would be to speed up the process for users).

Reply:
So if a customer can get the data into storage in the format we've recommended then we provide a script that does the import step for them? I like that. I think the only question is where is the data stored - if it's just in files then that's easy. Otherwise, we'd need to provide support for various data stores.

Reply:
I think we can just assume csv/json files - just pick whatever is easiest (don't need to spend too much time on it, but I figured us spending extra 15min might save everyone else spending 10min), if they have it in some other format they can make the conversion step beforehand. So they give us two files: events & persons with the expected format (the script can be in charts-clickhouse repo and could contain checks for input format, but that's low priority)

Reply:
I've created https://github.com/PostHog/meta/issues/13 to track this